tests/test_calc_error.py

```diff
@@ -0,0 +1,459 @@
+        self.assertEqual(
+            "Configurable cutoff must be some float value in the domain (0, 1]."  
+            " Received 10. Resetting value to 0.95",
+            str(cm.warning)
+        )
+
+        # Initializing parameters to very erroneous values       
+        m.parameters['qMax'] = 4000
+        keys = ['qMax']
+
+        # Before running estimate_params
+        with self.assertRaises(ValueError):
+            m.calc_error(simulated_results.times, simulated_results.inputs, simulated_results.outputs, dt=1)
+
+        m.estimate_params([(simulated_results.times, simulated_results.inputs, simulated_results.outputs)], keys, dt=0.5)
+
+        # After running estimate_params. Note that this would not change the outcome of the result
+        with self.assertRaises(ValueError):
+            m.calc_error(simulated_results.times, simulated_results.inputs, simulated_results.outputs, dt=1)    
+
+        # Running through various dt values that could work
+        for i in np.arange(0, 1, 0.1):
+            with self.assertRaises(ValueError):
+                m.calc_error(simulated_results.times, simulated_results.inputs, simulated_results.outputs, dt=i)
+
+        for i in range(2, 10):
+            with self.assertRaises(ValueError):
+                m.calc_error(simulated_results.times, simulated_results.inputs, simulated_results.outputs, dt=i)
+ 
+        # Creating duplicate model
+        m1 = BatteryElectroChemEOD()
+
+        orig_params = m1.parameters.copy()
+
+        # Much bigger parameter initialization
+        m.parameters['kp'] = m1.parameters['kp'] = 10000
+        m.parameters['kn'] = m1.parameters['kn'] = 1000 
+        m.parameters['qpMax'] = m1.parameters['qpMax'] = 4500
+        m.parameters['qMax'] = m1.parameters['qMax'] = 9000
+        keys = ['kp', 'kn', 'qpMax','qMax']
+
+        change_params = m1.parameters.copy()
+
+        simulated_results = m.simulate_to(2000, future_loading, **options)
+        m1_sim_results = m1.simulate_to(2000, future_loading, **options)
+
+        data = [(simulated_results.times, simulated_results.inputs, simulated_results.outputs)]
+        data_m1 = [(m1_sim_results.times, m1_sim_results.inputs, m1_sim_results.outputs)]
+
+        # Check out the warnings that are occurring here...
+        # They are being spammed almost. Increasing save_frequency increases spam
+        # Calling estimate_params does not change any of the parameters here because we are always accounting for exceptions...
+
+        m.estimate_params(data, keys, method='Powell')
+        m1.estimate_params(data_m1, keys, method='CG')
+
+        updated_params = m1.parameters.copy()
+
+        # Checking to make sure estimate_params actually changed values away from the original and to something else
+        self.assertEqual(change_params, updated_params)
+        self.assertNotEqual(orig_params, updated_params)
```

Comment:
This also doesn't really feel like a test of calc_error. But let's chat about it. I'll be curious to hear your perspective!